src/utils/TagUtils.ts

```diff
@@ -30,6 +30,10 @@ export default class TagUtils {
         // Going off local tags since it's our categories. Trying to find only changed tags and returning those
         // as a final object.
         Object.keys(localTags).forEach(key => {
+            if (remoteTags[key] === undefined) {
+                finalTags[key] = localTags[key];
+            }
```

Comment:
this check already exists here `if (remoteTags[key] && ....) {`

Reply:
think about the case where there aren't any remote tags yet

Reply:
Got it. How about a shorter version of it then?
```
Object.keys(localTags).forEach(key => {
      // only if user's tag value did not change, skip it
      if (localTags[key] === remoteTags[key]) {
         return;
      }
      finalTags[key] = localTags[key];
});
```

Reply:
Done.